### ISSUE TITLE
Clarify `reset()` behaviour when multiple things are being output

### ISSUE THREAD

**Issue by padenot** [w3c/webcodecs#755]
See https://github.com/web-platform-tests/interop/issues/614

Let's say I have an encoder that has quite a long latency, e.g. 16 frames.

If we enqueue 15 frames for encoding, call `flush()` and in output callback, call `reset()` on the encoder after e.g. the 8th callback, what should happen?

Firefox outputs the 15 frames, the `flush()` succeeds, and then the encoder is reset, because there's no prose to check that the encoder has been reset in the algorithm that calls the callback. But the description of the `reset()` method says that callbacks are not to continue (paraphrasing).

The intent might have been to _not_ call the callbacks after a reset, and we should make it explicit if that's the case.

**Comment by youennf** [comment]
Spec is indeed not very precise. It makes sense to me for `reset()` to cancel all previous tasks.
According https://wpt.fyi/results/webcodecs/video-encoder-flush.https.any.html?label=master&label=experimental&aligned, it seems Chrome and Safari agree here.

**Comment by padenot** [comment]
Yes but the point is that depending on the encoder being used, `reset()` hasn't been called when `flush()` is called, details are in the proposed change, there are comments.
